### PR TITLE
Add client support for SeriesGroupby unique, nsmallest, nlargest

### PR DESCRIPTION
**Pull request by naren-ponder**
Signed-off-by: Naren Krishna <naren@ponder.io>

## What do these changes do?

- [ ] passes `flake8 modin`
- [ ] passes `black --check modin`
- [ ] signed commit with `git commit -s` 
- [ ] Resolves #? 
- [ ] tests added and passing

### CODE REVIEW

File: modin/pandas/groupby.py [hunk 1]
```diff
@@ -873,8 +873,8 @@ def cumcount(self, ascending=True):
     def tail(self, n=5):
         return self._check_index(
             self._wrap_aggregation(
-                type(self._query_compiler).groupby_tail, 
-                agg_kwargs=dict(n=n), 
+                type(self._query_compiler).groupby_tail,
+                agg_kwargs=dict(n=n),
```
Comment (pyrito) [inline modin/pandas/groupby.py]: Are these lint changes?

File: modin/pandas/groupby.py [hunk 2]
```diff
@@ -1297,6 +1297,31 @@ def _iter(self):
...
+                numeric_only=False,
+            )
+        )
+
+    def nlargest(self, n=5, keep="first"):
+        return self._check_index(
+            self._wrap_aggregation(
+                type(self._query_compiler).groupby_nlargest,
+                agg_kwargs=dict(n=n, keep=keep),
+                numeric_only=True,
```
Comment (pyrito) [inline modin/pandas/groupby.py:1323]: does numeric_only consider booleans as well? It looks like bool dtypes work for `nlargest` and `nsmallest`.

File: modin/pandas/groupby.py (hunk 1 above)
Comment (naren-ponder) [reply modin/pandas/groupby.py]: Yes

File: modin/pandas/groupby.py (hunk 2 above)
Comment (naren-ponder) [reply modin/pandas/groupby.py:1323]: Hmm so this should be `False` then?

Comment (pyrito) [reply modin/pandas/groupby.py:1323]: @naren-ponder I'm not completely sure to be honest. I don't know if bool is considered as a numeric value?

Comment (pyrito) [reply modin/pandas/groupby.py:1323]: @naren-ponder it looks like bool is considered as numeric values. Can you just double check this with other dtypes? Looks like strings aren't supported, so I think this is ok.

**Review by pyrito — APPROVED** [review]

**Pull request merged**